impeller/renderer/backend/vulkan/context_vk.cc

```diff
@@ -377,10 +377,6 @@ void ContextVK::Setup(Settings settings) {
   ///
   auto fence_waiter =
       std::shared_ptr<FenceWaiterVK>(new FenceWaiterVK(device_holder));
```

Comment:
You also need to add fence waiter and resource manager teardown into the shutdown method, right? otherwise the dtor order is still incorrect

Reply:
Yeah. I still planned to do this in https://github.com/flutter/engine/pull/45654, I could do it here but I was trying to minimize changes to "what passes on device and CI".

Reply:
I think you should do it here and then try to get CI enabled with validation

Reply:
That is, we should fix the known errors before we try to land the larger performance/feature changes?

Reply:
I would consider this PR fixing a known error, not a performance/feature change but I have no problem adding the Shutdown fix as well - the only reason I was hesitant is without the ResourceManager change I couldn't observe a problem with shutdown (it's also possible this fence waiter PR "makes stuff work").

That being said I have no problem adding the 2 lines to context_vk.cc in this PR. Wdut?

Reply:
Ah you mean the resource manager being a performance improvement, that's fair.

Reply:
If you want, you could patch in the change to add validation and see if it passes without that fix.

Reply:
Yeah let me do that locally at least. Good suggestion!

Reply:
At least locally, I don't seem to need any other changes.

As discussed on chat, given this is already a risky change, I'm trying not to introduce more chances at changing behavior - happy to still include the termination fixes in a follow-up PR (w/ or w/o the ResourceManager optimizations)